Fixed order of statements RaidenService's constructor

hackaugusto opened the pull request:
Merge after: #953

- To avoid races, moved the protocol start to the end of the
constructor. Messages may fail to be processed if the node is partially
instantiated.
- Moved the default registry registration out of the database guard.
- Unconditonally waiting for the endpoint registration.

LefterisJP commented on raiden/message_handler.py:
```diff
@@ -47,19 +45,15 @@ def __init__(self, raiden):
-
-        elif cmdid == messages.PING:
-            pass
+        # ack and ping messages are not forwarded to the handler
```
You can just remove the commented out code and leave the initial comment.

hackaugusto replied:
done

LefterisJP commented on raiden/network/protocol.py:
```diff
@@ -655,24 +656,46 @@ def receive(self, data):
+                    'PING RECEIVED',
+                    node=pex(self.raiden.address),
+                    echohash=pex(echohash),
+                    message=message,
```
can you also add a `sender=message.sender` here?

hackaugusto replied:
done

LefterisJP commented on raiden/network/protocol.py:
```diff
@@ -655,24 +656,46 @@ def receive(self, data):
+                )
+
+            # ping handler does nothing
+            # self.raiden.on_message(message, echohash)
```
Just remove this. I also saw this while working on the ping problem yesterday.

hackaugusto commented:
> since it's hard to compare the diff: did you do any logical changes I should be aware of (other than changing the order)?

So this is what the PR did:
- change the RaidenService constructor initialization order. The goal is:
  -  Avoid handling packets with a partially initialized node
  - Avoid missing blockchain events because of races with the alarm task / event listener installation.
  - Changes to the registration, because the transaction may fail it's important to not start the node before it finishes.
- changed a bit of the logging from the protocol
  - This was to make debugging easier for me, the fixes were added back to previous commits
- added a test for the direct transfer with an offline node
  - This is a bit unrelated but I sneaked the test in (I was writing this test when I realized that I needed restart the node, and was doing much the same thing)
  - This is the reason for the start/stop methods.

Now, I'm *assuming* that the alarm task is actually working, I did not go through all code paths and tinkered about all possible races. The initialization of the protocol on the other hand should be enough to fix errors with a partially initialized node.
